presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveBasicTableStatistics.java

```diff
@@ -0,0 +1,473 @@
+            assertThatStatisticsAreNonZero(partitionStatisticsFirstInsert);
+            assertThat(partitionStatisticsFirstInsert.getNumRows().getAsLong()).isEqualTo(5);
+
+            insertNationData(onPresto(), tableName);
+
+            BasicStatistics tableStatisticsSecondInsert = getBasicStatisticsForTable(onHive(), tableName);
+            assertThat(tableStatisticsSecondInsert.getNumRows().getAsLong()).isEqualTo(50);
+            assertThatStatisticsValuesHaveIncreased(tableStatisticsFirstInsert, tableStatisticsSecondInsert);
+            assertThat(tableStatisticsSecondInsert.getNumPartitions().getAsLong()).isEqualTo(5);
+
+            BasicStatistics statisticsSecondInsert = getBasicStatisticsForPartition(onHive(), tableName, "n_regionkey=3");
+            assertThat(statisticsSecondInsert.getNumRows().getAsLong()).isEqualTo(10);
+            assertThatStatisticsValuesHaveIncreased(partitionStatisticsFirstInsert, statisticsSecondInsert);
+        }
+        finally {
+            onPresto().executeQuery(format("DROP TABLE IF EXISTS %s", tableName));
+        }
+    }
+
+    @Test(groups = {HIVE_CONNECTOR, HIVE_TABLE_STATISTICS})
+    public void testInsertBucketed()
+    {
+        String tableName = "test_basic_statistics_bucketed_insert_presto";
+
+        onPresto().executeQuery(format("DROP TABLE IF EXISTS %s", tableName));
+        onPresto().executeQuery(format("" +
+                "CREATE TABLE %s " +
+                "WITH ( " +
+                "   bucketed_by = ARRAY['n_nationkey'], " +
+                "   bucket_count = 50 " +
+                ") " +
+                "AS " +
+                "SELECT n_nationkey, n_name, n_comment, n_regionkey " +
+                "FROM nation", tableName));
+
+        try {
+            BasicStatistics statisticsAfterCreate = getBasicStatisticsForTable(onHive(), tableName);
+            assertThatStatisticsArePresent(statisticsAfterCreate);
+            assertThatStatisticsAreNonZero(statisticsAfterCreate);
+            assertThat(statisticsAfterCreate.getNumRows().getAsLong()).isEqualTo(25);
+            assertThat(statisticsAfterCreate.getNumFiles().getAsLong()).isEqualTo(50);
+            assertThat(statisticsAfterCreate.getNumPartitions().getAsLong()).isEqualTo(0);
+
+            // Insert into bucketed unpartitioned table is unsupported
+            // Once supported - uncomment assertions below
+            assertThatThrownBy(() -> insertNationData(onPresto(), tableName))
+                    .hasMessageContaining("Cannot insert into bucketed unpartitioned Hive table");
+
+            // BasicStatistics statisticsAfterInsert = getBasicStatisticsForTable(onHive(), tableName);
+            // assertThatStatisticsValuesHaveIncreased(statisticsAfterCreate, statisticsAfterInsert);
+            // assertThat(statisticsAfterInsert.getNumRows().getAsLong()).isEqualTo(50);
+            // assertThat(statisticsAfterCreate.getNumFiles().getAsLong()).isEqualTo(100);
```

Comment:
Technically, you could uncomment these lines (with expected values := un-updated values)